features/step_definitions/framework_application_steps.rb

```diff
@@ -0,0 +1,49 @@
+Then /^I am on the #{MAYBE_VAR} page for that framework application$/ do |page_title|
+  page_title.sub! "framework", @framework['name']
+  step "I am on the '#{page_title}' page"
+end
+
+Then /^I click #{MAYBE_VAR} link for that framework application$/ do |link_title|
+  link_title.sub! "framework", @framework['name']
+  step "I click a link with text '#{link_title}'"
+end
+
+Then(/^I answer all questions on that page$/) do
+  page_header_at_start = page.all(:xpath, "//h1")[0].text
+  edit_links = page.all(:xpath, "//p[@class='summary-item-top-level-action']/a[text()='Edit']")
+  edit_links[0].click
+  page_header = nil
+  until page_header_at_start == page.all(:xpath, "//h1")[0].text
+    if page_header == page.all(:xpath, "//h1")[0].text
+      options = get_answers_for_validated_questions
+      answer = fill_form with: options
+    else
+      page_header = page.all(:xpath, "//h1")[0].text
+      answer = fill_form
+    end
+    merge_fields_and_print_answers(answer)
+    find_and_click_submit_button
+  end
+end
+
+Then(/^I submit a service for each lot$/) do
+  lots_links_length = find_elements_by_xpath("//ul[@class='browse-list']//a").length
+  index = 0
+  lots_links_length.times do
```

Comment:
Can we use a standard `each` here?

```
find_elements_by_xpath("//ul[@class='browse-list']//a").each do |link|
  link.click
  ...
end
```
If not then `each_with_index` is available:

```
find_elements_by_xpath("//ul[@class='browse-list']//a").each_with_index do |link, index|
  link = find_elements_by_xpath("//ul[@class='browse-list']//a")[index]
  link.click
  ...
end
```

Reply:
The thing is, after you click through once, it's not the same element anymore. That's why the link has to be freshly discovered each time.